aiida_quantumespresso/parsers/base.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+"""Defines a `Parser` base class for `aiida-quantumespresso`.
+
+All `Parser` implementations in `aiida-quantumespresso` must use this base class, not `aiida.parsers.Parser`.
+"""
+from __future__ import absolute_import
+
+from aiida.parsers import Parser as _BaseParser
+
+__all__ = ('Parser',)
+
+
+class Parser(_BaseParser):  # pylint: disable=abstract-method
+    """Custom `Parser` class for `aiida-quantumespresso` parser implementations."""
+
+    def emit_logs(self, logging_dictionaries, ignore=None):
+        """Emit the messages in one or multiple "log dictionaries" through the logger of the parser.
+
+        A log dictionary is expected to have the following structure: each key must correspond to a log level of the
+        python logging module, e.g. `error` or `warning` and its values must be a list of string messages. The method
+        will loop over all log dictionaries and emit the messages it contains with the log level indicated by the key.
+
+        Example log dictionary structure::
+
+            logs = {
+                'warning': ['Could not parse the `etot_threshold` variable from the stdout.'],
+                'error': ['Self-consistency was not achieved']
+            }
+
+        :param logging_dictionaries: log dictionaries
+        :param ignore: list of log messages to ignore
+        """
+        ignore = ignore or []
+
+        if not isinstance(logging_dictionaries, (list, tuple)):
+            logging_dictionaries = [logging_dictionaries]
+
+        for logs in logging_dictionaries:
+            for level, messages in logs.items():
+                for message in messages:
+
+                    if message is None:
+                        continue
+
+                    stripped = message.strip()
+
+                    if not stripped or stripped in ignore:
+                        continue
+
+                    try:
+                        getattr(self.logger, level)(stripped)
+                    except AttributeError:
```

Comment:
In the case where `level` does not correspond to a known logging level, wouldn't it be safer to fall back on e.g. `warning`?

Is there a concrete use case where you want to suppress the logging, or is this just to not fail on typos?

Reply:
In principle this would be a programming error, because the programmer is supposed to use a dictionary that is created by `get_logging_container` which only supports `warning` and `error` keys for the moment. It is all a bit flimsy at the moment and just relies on the developers doing the right thing, but since that is basically me, I think it is fine for now.